### ISSUE TITLE
Error on WPCS installation with composer 2 

### ISSUE THREAD

**Issue by hmartinez69** [WordPress/WordPress-Coding-Standards#1979]
## Bug Description

At installation with using command line `composer create-project wp-coding-standards/wpcs --no-dev` with a MacOS computer with Composer version 2.0.11 realeased on 2021-02-24 14:57:23, I had an error of installation
## Minimal Code Snippet

I don't have any code to place because it's before to coding

For bugs with fixers: How was the code fixed? How did you expect the code to be fixed?

## Error Code

```
% composer create-project wp-coding-standards/wpcs --no-dev
Creating a "wp-coding-standards/wpcs" project at "./wpcs"
Installing wp-coding-standards/wpcs (2.3.0)
  - Downloading wp-coding-standards/wpcs (2.3.0)
  - Installing wp-coding-standards/wpcs (2.3.0): Extracting archive
Created project in /Users/admin/Documents/dev/wpcs
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - dealerdirect/phpcodesniffer-composer-installer[v0.5.0, ..., v0.6.2] require composer-plugin-api ^1.0 -> found composer-plugin-api[2.0.0] but it does not match the constraint.
    - Root composer.json requires dealerdirect/phpcodesniffer-composer-installer ^0.5 || ^0.6 -> satisfiable by dealerdirect/phpcodesniffer-composer-installer[v0.5.0, v0.6.0, v0.6.1, v0.6.2].

You are using Composer 2, which some of your plugins seem to be incompatible with. Make sure you update your plugins or report a plugin-issue to ask them to support Composer 2.
Running update with --no-dev does not mean require-dev is ignored, it just means the packages will not be installed. If dev requirements are blocking the update you have to resolve those problems.
```

## Environment

| Question               | Answer
| ------------------------| -------
| PHP version             |7.3.24
| PHP_CodeSniffer version | x.y.z
| WPCS version            | x.y.z
| WPCS install type       | e.g. Composer global, Composer project local, git clone, other (please expand)
| IDE (if relevant)       | Name and version e.g. PhpStorm 2018.2.2

## Tested Against `develop` branch?
- [x] I have verified the issue still exists in the `develop` branch of WPCS.

**Comment by jrfnl** [comment]
@hmartinez69 Sorry about that. We had hoped to release the new version already, but progress has been slow due to other commitments of contributors.

There is basically a combination of two issues here:
1. The installation instructions are kind of outdated as the recommended one nowadays would be to use `composer [global] require ...`
2. One of the `dev` dependencies of WPCS is "fixed" at a version which is only compatible with Composer 1.0. The dependency has released a newer version which is compatible with both Composer 1.x as well as 2.x, but the version constraints in WPCS 2.3.0 do not allow for it.

Do you want WPCS to be available globally to all projects on your system or do you want it to be available to a specific project ?

If for a project only, try this:
* Navigate to the project root directory
* Run
    ```
    composer require dealerdirect/phpcodesniffer-composer-installer --update-no-dev
    composer require wordpress/wpcs --update-no-dev
    ```

If for all projects on your system:
* Run:
    ```
    composer global require dealerdirect/phpcodesniffer-composer-installer --update-no-dev
    composer global require wordpress/wpcs --update-no-dev
    ```
* And make sure the path where Composer installs in is in your OS system path.
    To find out where Composer installs `global` packages, run `composer global info`

Hope that helps.

**Comment by hmartinez69** [comment]
Thanks @jrfnl for your answer and your informations.
I need to install WPCS for all projects so I've tried your steps. First step was good.
But for second step, I had that on my command line 

```
% composer global require wordpress/wpcs --update-no-dev
Changed current directory to /Users/admin/.composer

                                                                                                                                                                                                         
  [InvalidArgumentException]                                                                                                                                                                             
  Could not find a matching version of package wordpress/wpcs. Check the package spelling, your version constraint and that the package is available in a stability which matches your minimum-stabilit  
  y (stable).                                                                                                                                                                                            
                                                                                                                                                                                                         

require [--dev] [--dry-run] [--prefer-source] [--prefer-dist] [--fixed] [--no-suggest] [--no-progress] [--no-update] [--no-install] [--no-scripts] [--update-no-dev] [-w|--update-with-dependencies] [-W|--update-with-all-dependencies] [--with-dependencies] [--with-all-dependencies] [--ignore-platform-req IGNORE-PLATFORM-REQ] [--ignore-platform-reqs] [--prefer-stable] [--prefer-lowest] [--sort-packages] [-o|--optimize-autoloader] [-a|--classmap-authoritative] [--apcu-autoloader] [--apcu-autoloader-prefix APCU-AUTOLOADER-PREFIX] [--] [<packages>]...
```

How can I fix that ?

**Comment by jrfnl** [comment]
@hmartinez69 Sorry, I had a typo in my original answer and edited it as soon as I posted it. It should be `composer global require wp-coding-standards/wpcs --update-no-dev`.

**Comment by hmartinez69** [comment]
Ok sorry, no problem ;)
I've done this command line with fixed package. But phpcs or wpcs doesn't work.
Sorry for my many questions, but how can I fix that ? I need to add a PATH variable as phpcs or wpcs ?

Thanks for all

**Comment by hmartinez69** [comment]
Sorry, @jrfnl I've fixed that ! All is good ! 
Thanks for all

**Comment by jrfnl** [comment]
@hmartinez69 Glad to hear you got it working. Enjoy!

**Comment by janw-me** [comment]
Hi,
I've had the same problem and fix it with @jrfnl suggestions.
Will this be fixed with wpcs v3.0.0? Or is there an other issue/pull/ect I should follow for the definitive fix?

**Comment by jrfnl** [comment]
@janw-me The install instructions will be completely updated in the README for WPCS 3.0.0.
